EVG-6879 / EVG-6900 update schema for historical test stats

jimoleary opened the pull request.

tychoish commented on model/testresult/testresult.go:
```diff
@@ -17,22 +19,32 @@ const (
+	Requester            string    `bson:"r" json:"r"`
+	DisplayName          string    `bson:"display_name" json:"display_name"`
+	ExecutionDisplayName string    `bson:"execution_display_name,omitempty" json:"execution_display_name,omitempty"`
+	CreateTime           time.Time `bson:"create_time" json:"create_time"`
```
can we also have a complete time so we can migrate away from the floats?

jimoleary replied:
@tychoish 
I think that this change request is out of scope for this work.

The start time and end time will continue to be used regardless.

jimoleary replied:
It's completely trivial, but on the other hand it is also completely unrelated to this fix.

tychoish replied:
Please do it anyway. It doesn't make sense to change the schema multiple times.

tychoish replied:
The Start/End Fields should have time.Time equivalents in the new schema.

jimoleary commented:
Updated, I called the collection fields start_date and end_date just so that it is not confusing.

jimoleary commented:
create_time is the task_create_time. start_date is the test_start_time.

StartTime (start in the database) and EndTime (end) need to remain the same as they are already in use.

But I can change start_date, end_data and create_time to make them nore obvious. 

AFAICS the failing tests are unrelated to these changes but I'll look into it.

jimoleary commented:
Rebased onto master and the tests that now fail have changed. There is a failure due to auth in test-rest-route and a setup failure in test-cloud, neither of which are related to this set of changes.

@tychoish I've changed the struct field names (that can be changed). Hopefully they are more clear now.
